internal/pkg/utils/winpol/windows_policy.go

```diff
@@ -85,13 +100,89 @@ func CalculateEndpointPolicies(
 		encoded, err := json.Marshal(dict)
 		if err != nil {
 			logger.WithError(err).Error("Failed to add outbound NAT exclusion.")
-			return nil, err
+			return nil, nil, err
 		}
 
-		outputPols = append(outputPols, json.RawMessage(encoded))
+		outputV1Pols = append(outputV1Pols, json.RawMessage(encoded))
+		// Get v2 policy.
+		v2Pol, err := convertToHcnEndpointPolicy(dict)
+		if err != nil {
+			logger.WithError(err).Error("Failed to convert endpoint policy to HCN endpoint policy.")
+			return nil, nil, err
+		}
+		outputV2Pols = append(outputV2Pols, v2Pol)
 	}
 
-	return outputPols, nil
+	return outputV1Pols, outputV2Pols, nil
+}
+
+// convertToHcnEndpointPolicy converts a map representing the raw data of a V1
+// policy and converts it to an HCN endpoint policy.
+//
+// For example, we convert from raw JSON like:
+//
+// {
+//   "Type":  "OutBoundNAT",
+//   "ExceptionList":  [
+//     "10.96.0.0/12",
+//     "192.168.0.0/16"
+//   ]
+// }
+//
+// to:
+//
+// hcn.EndpointPolicy{
+//   Type: hcn.OutBoundNAT,
+//   Settings: json.RawMessage(
+//     []byte(`{"ExceptionList":["10.96.0.0/12","192.168.0.0/16"]}`),
+//   ),
+// }
+func convertToHcnEndpointPolicy(policy map[string]interface{}) (hcn.EndpointPolicy, error) {
+	hcnPolicy := hcn.EndpointPolicy{}
+
+	policyType := policy["Type"].(string)
+	// Get v2 policy type.
+	v2PolicyType, err := getHcnEndpointPolicyType(policyType)
+	if err != nil {
+		return hcnPolicy, fmt.Errorf("Invalid endpoint policy type")
```

Comment:
Ahh okay, I did this because `hcnPolicy` is `hcn.EndpointPolicy` and not a pointer. The policies object is used later on to set a `hcn.HostComputeEndpoint`, which takes`[]hcn.EndpointPolicy{}`. I don't know if it's worth turning `hcnPolicy` into a pointer only to dereference it later.